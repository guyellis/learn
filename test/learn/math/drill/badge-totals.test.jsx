
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const BadgeTotals = require('../../../../src/learn/math/drill/badge-totals');

const muiTheme = getMuiTheme(lightBaseTheme);

test('Badge totals should be rendered', () => {
  const component = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <BadgeTotals totals={[1, 2, 3]} />
    </MuiThemeProvider>);
  const badgeTotals = component.toJSON();
  expect(badgeTotals).toMatchSnapshot();
});
