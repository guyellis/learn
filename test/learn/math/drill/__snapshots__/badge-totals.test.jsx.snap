// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge totals should be rendered 1`] = `
<div>
  <div>
    <button
      className="MuiButtonBase-root MuiFab-root MuiFab-sizeMedium MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "fontSize": "1.5em",
          "margin": "5px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        1
      </span>
    </button>
    <span>
      Gold Badge(s) - 2 seconds or less (per question)
    </span>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiFab-root MuiFab-sizeMedium MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "fontSize": "1.5em",
          "margin": "5px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        2
      </span>
    </button>
    <span>
      Silver Badge(s) - between 2 and 3 seconds (per question)
    </span>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiFab-root MuiFab-sizeMedium MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "fontSize": "1.5em",
          "margin": "5px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        3
      </span>
    </button>
    <span>
      Bronze Badge(s) - between 3 and 4 seconds (per question)
    </span>
  </div>
</div>
`;
